lib/cogctl/actions/groups/delete.ex

```diff
@@ -1,25 +1,33 @@
 defmodule Cogctl.Actions.Groups.Delete do
   use Cogctl.Action, "groups delete"
 
+  alias CogApi.HTTP.Client
+
   def option_spec do
     [{:group, :undefined, :undefined, {:string, :undefined}, 'Group name (required)'}]
   end
 
-  def run(options, _args, _config, endpoint) do
-    with_authentication(endpoint,
-                        &do_delete(&1, :proplists.get_value(:group, options)))
+  def run(options, _args, _config, %{token: nil}=endpoint) do
+    with_authentication(endpoint, &run(options, nil, nil, &1))
   end
 
-  defp do_delete(_endpoint, :undefined) do
-    display_arguments_error
+  def run(options, _args, _config, endpoint) do
+    group_name = :proplists.get_value(:group, options)
+    case Client.group_find(endpoint, name: group_name) do
+      {:ok, group} ->
+        do_delete(endpoint, group)
+      _ ->
+        display_error("Unable to find group named #{group_name}")
+    end
```

Comment:
Maybe the find_group can go into the Cogctl.Actions.Group module and then be imported in to use in `do_delete` as well as the other functions since it looks like it's the same code.